pkg/types/types.go

```diff
@@ -270,14 +275,32 @@ func (s byStepTime) Less(i, j int) bool {
 	return s[i].StepTime < s[j].StepTime
 }
 
-func mergeMetrics(metrics []Metric, mismatchCheck bool) (metric Metric, mismatches int) {
+func getPointMajorityValue(values []float64) (majorityValue float64, majorityCount int) {
+	valuesCount := len(values)
+	if valuesCount == 0 {
+		return 0, 0
+	}
+
+	valuesToCount := make(map[float64]int, valuesCount)
+	majorityValue = values[0]
+	for _, val := range values {
+		valuesToCount[val]++
```

Comment:
I assume this does simple floats comparison. It is unsafe to do byte-by-byte floats comparison as the representation may differ for the same number. Check this out https://github.com/golang/go/issues/966 for example.

Reply:
Nice catch. I'll change the implementation.